### PR TITLE
76 sprite class

### PR DESCRIPTION
**Pull request by michaeljscrockett**
Sprites are now all subclasses of `Sprite`, which has some useful methods like `flipHorizontal` and `setAlign`.

The only issue is that `TimeBlocks` is a little buggy. It can likely be rewritten using `imposeBlocks` from `stringManipulation.js`

### CODE REVIEW

**Comment by michaeljscrockett** [comment]
Here's some pictures of the horizontal flip working because they look cool:
Screenshot 2023-06-15 at 8 04 17 PM Screenshot 2023-06-15 at 8 06 05 PM

**Comment by drnikki** [comment]
@mollyinkilos look at this!

**Comment by drnikki** [comment]
@mollyinkilos look at this!

**Comment by mollyinkilos** [comment]
Hell yeah!

File: js/sprite/Sprite.js [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+     */
+    constructor(spriteText) {
+        this.spriteRows = typeof spriteText === 'string' ? spriteText.split(br) : spriteText;
+    }
+
+    /**
+     * finds the width of the sprite
+     * @returns {{width: number, start: number, end: number}}
+     */
+    #findWidth(){
```
Comment (drnikki) [inline js/sprite/Sprite.js:37]: Why do some properties and functions start with a #?  I'm not familiar with this convention.

File: js/sprite/Sprite.js [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+                // find the largest ending index
+                Math.max(end, currEnd),
+            ]},
+            [Sprite.receiptWidth, 0] // initial start and end
+        )
+        return {width: end - start, start, end};
+    }
+
+
+    toString() {
```
Comment (drnikki) [inline js/sprite/Sprite.js:57]: needs a function docblock

**Review by drnikki — COMMENTED** [review]
This looks SO GOOD - like seriously great work!!  (Was it fun!!).

File: js/sprite/Sprite.js (hunk 1 above)
Comment (michaeljscrockett) [reply js/sprite/Sprite.js:37]: It makes them private properties/methods. I suppose it doesn't really add anything, so I can remove them if that makes them more clear.

**Comment by michaeljscrockett** [comment]
> This looks SO GOOD - like seriously great work!! (Was it fun!!).
> 
> Aside from docblock updates, this is ready to merge. Feel free to merge it after those changes come in.

Yes, it was a lot of fun!

File: js/sprite/Sprite.js [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+        this.spriteRows = this.spriteRows.map(row => {
+            //find length, adjusted based on number of non-breaking spaces
+            const spaceCount = (row.match(/&nbsp;/g) || []).length;
+            const length = row.length - (5*spaceCount);
+            const padding = '&nbsp;'.repeat(Math.max(Sprite.receiptWidth - length, 0));
+
+            const flippedRow = row
+                .split('').reverse()  // flipped row with spaces and brackets backwards
+                .map(char => {  // flip the directed characters
+                    switch (char){
```
[the remark this answers is not in the document]
Comment (michaeljscrockett) [reply js/sprite/Sprite.js:94]: Thanks!

**Comment by michaeljscrockett** [comment]
@drnikki The PR is failing a test because it has an octal literal somewhere (a number padded on the left with a `0`). The problem is that I have no idea how to find it without searching every single file.

**Comment by drnikki** [comment]
Yes!  The Linter is not *quite* where I'd like it to be, so I disabled actions for today (but will try again tomorrow!)

**Pull request merged**